elftools/elf/segments.py

```diff
@@ -119,11 +120,20 @@ def iter_notes(self):
             self.stream.seek(offset)
             # n_namesz is 4-byte aligned.
             disk_namesz = roundup(note['n_namesz'], 2)
+            # skip parsing truncated notes.
+            if offset+disk_namesz+roundup(note['n_descsz'], 2) > end:
+                break;
             note['n_name'] = bytes2str(
                 CString('').parse(self.stream.read(disk_namesz)))
             offset += disk_namesz
 
             desc_data = bytes2str(self.stream.read(note['n_descsz']))
+            if note['n_name'] == 'GNU':                                       
```

Comment:
Thanks for the detailed explanation!

So when enums are embedded in construct, as was the case before your change - they *are* properly "inverted".

Given that you no longer use ENUM_NOTE_N_TYPE in structs.py, how about moving it out of enums (these are specifically for enums embedded in structs) and just put it somewhere in the note segment where you can then invert the map properly (it's not used elsewhere, is it?)

And please put all of this in a comment somewhere explaining this special treatment of notes - I wasn't aware of this and it will be good to keep the knowledge with the code.